src/app/dashboard/Dashboard.tsx

```diff
@@ -1,25 +1,46 @@
-import { Grid } from '@mui/material';
+import { Grid, Typography } from '@mui/material';
 
 import { styles } from './Dashboard.styles';
 import { RankingCard } from './rankingCard/RankingCard';
-import { RANKING_MOCK } from 'tests/mock/rankingMock';
 import { RankingTable } from './rankingTable/RankingTable';
+import { DashboardProps } from './Dashboard.types';
+import { Card, Spinner } from 'ui';
+import { useLocale } from 'hooks';
+
+export const Dashboard = ({ isLoading, isError, data }: DashboardProps) => {
+  const { formatMessage } = useLocale();
 
-export const Dashboard = () => {
   return (
     <>
-      <Grid container spacing={2} sx={styles.container} columns={{ xs: 6, sm: 12, md: 12 }}>
-        {RANKING_MOCK.sort((a, b) => b.score - a.score).map((rank, index) => (
-          <Grid key={rank.name} item xs={styles.item.width}>
-            <RankingCard
-              name={rank.name}
-              score={rank.score}
-              rank={RANKING_MOCK[0].score !== RANKING_MOCK[1].score ? index + 1 : 1}
-            />
+      {isLoading && (
```

Comment:
To by można było zrobić jako osobny return, tzn 
```
if (isLoading) {
  return (
    <Card>
      <Spinner />
    </Card>
  );
}
```
To samo z errorem. W taki sposób logika wyświetlenia kolejnych elementów będzie dużo bardziej czytelna. No i nie będziesz musiał sprawdzać, czy `data !== undefined`

Reply:
Hmmm, zgadzam się i tak zrobiłem, tylko, że TypeScript i tak mi podkreśla `data` i wyrzuca `Object is possibly 'undefined'.` - nie widzi zależności, że jeżeli `!isLoading && !isError` to `data !== undefined`. Jest jakieś obejście tego problemu?

Reply:
Albo na samym początku zrobić ifa:
```
if (!isLoading && !isError && !data) {
  return null;
}
```
wg uzania

Reply:
Hehe, zanim przeczytałem ten komentarz zrobiłem w drugim ifie  `isError || !data`. A odpowiadając na pytanie `isLoading || !data` w pierwszym ifie da `true` również w przypadku kiedy będzie error, czyli `if(isError)` nigdy się nie wykona.